scripts/biomedical/VirtualMetabolicHuman/hmdb_extract_csv.py

```diff
@@ -0,0 +1,216 @@
+"""
+Author: Suhana Bedi
+Date: 07/10/2021
+Name: hmdb_extract.py
+Description: Extracts the relevant tags and their corresponding 
+information from the hmdb_metabolites xml file and converts it
+into a cleaner and more readable csv file.
+@file_output: csv output file 
+"""
+
+import pandas as pd
+import numpy as np
+import datacommons as dc
+import sys
+import os
+import csv
+from io import StringIO
+from lxml import etree
+"""
+Extracts relevant information from xml version of metabolites
+file and converts it into a csv
+Args:
+    name = name of the xml file
+    file = name of the resulting csv file 
+Returns:
+    output csv 
+Source : http://www.metabolomics-forum.com/index.php?topic=1588.0
+"""
+
+
+def hmdbextract(name, file):
+    ns = {'hmdb': 'http://www.hmdb.ca'}
+    context = etree.iterparse(name, tag='{http://www.hmdb.ca}metabolite')
+    csvfile = open(file, 'w')
+    fieldnames = [
+        'accession', 'monisotopic_molecular_weight', 'iupac_name', 'name',
+        'chemical_formula', 'InChIKey', 'cas_registry_number', 'smiles',
+        'drugbank', 'chebi_id', 'pubchem', 'phenol_explorer_compound_id',
+        'food', 'knapsack', 'chemspider', 'kegg', 'meta_cyc', 'bigg',
+        'metlin_id', 'pdb_id', 'logpexp', 'kingdom', 'direct_parent',
+        'super_class', 'class', 'sub_class', 'molecular_framework', 'vmh_id'
+    ]
+    writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+    writer.writeheader()
+    for event, elem in context:
+
+        accession = elem.xpath('hmdb:accession/text()', namespaces=ns)[0]
+        try:
+            monisotopic_molecular_weight = elem.xpath(
+                'hmdb:monisotopic_molecular_weight/text()', namespaces=ns)[0]
+        except:
+            monisotopic_molecular_weight = 'NA'
```

Comment:
again, since this is a repetitive pattern - a helper function would help a lot here. also, avoid try/except blocks if possible - in this case, is it sufficient to just test the length of the returned element list?